ICSharpCode.CodeConverter/CSharp/NodesVisitor.cs

```diff
@@ -1855,18 +1855,31 @@ public override CSharpSyntaxNode VisitIdentifierName(VBSyntax.IdentifierNameSynt
                                                     || node.Parent is VBSyntax.MemberAccessExpressionSyntax maes && maes.Expression == node
                                                     || node.Parent is VBSyntax.QualifiedNameSyntax qns && qns.Left == node
                     ? QualifyNode(node, identifier) : identifier;
-                return AddEmptyArgumentListIfImplicit(node, qualifiedIdentifier);
+
+                var withArgList = AddEmptyArgumentListIfImplicit(node, qualifiedIdentifier, out bool addedArgumentList);
+                if (!addedArgumentList && !node.Parent.IsKind(VBasic.SyntaxKind.InvocationExpression)) {
```

Comment:
Probably need to watch out for `AddressOf Foo` as well.
More nebulous: Technically `NameOf(Foo)` probably shouldn't change in case someone depends on it.
I wonder if we can use the symbol type instead to check for anything with that name that isn't equal to the method's symbol? It might avoid the need for the special case below too where you need to know if it was turned into an invocation.

Reply:
I've updated this to use the symbol information instead - this is far nicer.

I've still special-cased `NameOf` - it's semantically different, but produces the same results as in VB. I don't know if this is desired or not.